test/core/test_html_structure.rb

```diff
@@ -115,6 +115,38 @@ def test_render_with_custom_shortcut_and_additional_attrs
                 source, shortcut: {'^' => {tag: 'script', attr: 'data-binding', additional_attrs: { type: "application/json" }}}
   end
 
+  def test_render_with_custom_lambda_shortcut
+    source = %q{
+~foo Hello
+}
+    assert_html '<div class="styled-foo">Hello</div>',
+                source, shortcut: {'~' => {attr: ->(v) {{class: "styled-#{v}"}}}}
+  end
+
+  def test_render_with_custom_lambda_shortcut_and_multiple_values
+    source = %q{
+~foo~bar Hello
+}
+    assert_html '<div class="styled-foo styled-bar">Hello</div>',
+                source, shortcut: {'~' => {attr: ->(v) {{class: "styled-#{v}"}}}}
+  end
+
+  def test_render_with_custom_lambda_shortcut_and_existing_class
+    source = %q{
+~foo.baz Hello
+}
+    assert_html '<div class="styled-foo baz">Hello</div>',
+                source, shortcut: {'~' => {attr: ->(v) {{class: "styled-#{v}"}}}, '.' => {attr: 'class'}}
```

Comment:
Is there a way to keep the default shortcuts (`.` and `#`) while adding new custom ones, without having to redefine the defaults?

Reply:
In order to add new custom shortcuts, you have to mutate the hash.

Reply:
How can I access the configured shortcuts Hash then?

Reply:
Oh sorry about that. I'll update this when I have the chance. Apart from this, everything looking okay for merging?

Reply:
I updated my tests to use `Slim::Parser.options[:shortcut].update` instead of the `shortcut` argument.